src/schema/posts.ts

```diff
@@ -527,6 +534,20 @@ export const typeDefs = /* GraphQL */ `
       id: ID!
     ): EmptyResponse @auth
 
+    """
+    Fetch external link's title and image preview
+    """
+    checkLinkPreview(
+      """
+      Source the user is planning to post to
+      """
+      sourceId: ID!
```

Comment:
Why the source id is relevant here?
Potentially we can use this even for comments and other links in the app